### PR TITLE
API-39: Add integration tests on metric & number attributes validation

### PR DESCRIPTION
**Pull request by phaseinducer**

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/NumberIntegration.php [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        );
+
+        $violations = $this->validateAttribute($attribute);
+
+        $this->assertCount(1, $violations);
+        $this->assertSame('This value should be null.', $violations->get(0)->getMessage());
+        $this->assertSame('dateMax', $violations->get(0)->getPropertyPath());
+    }
+
+    public function testMetricShouldNotHaveAMaxFileSize()
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/NumberIntegration.php]: testNumber

File: src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/NumberIntegration.php [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        );
+
+        $violations = $this->validateAttribute($attribute);
+
+        $this->assertCount(1, $violations);
+        $this->assertSame('This value should be null.', $violations->get(0)->getMessage());
+        $this->assertSame('metricFamily', $violations->get(0)->getPropertyPath());
+    }
+
+    public function assertDoesNotHaveADefaultMetricUnit($type)
```
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/NumberIntegration.php]: copy/paste -_-

File: src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/GlobalConstraintsIntegration.php [hunk 3]
```diff
@@ -257,6 +260,14 @@ public function testReservedCode()
         $this->assertSame('code', $violations->get(0)->getPropertyPath());
     }
 
+    public function reservedCodesProvider()
+    {
+        return [
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/GlobalConstraintsIntegration.php]: not possible to get those codes dynamically ?

Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Validation/Attribute/GlobalConstraintsIntegration.php]: Nope, it's a pattern in validation conf so...

**Pull request merged**